app/components/dashboard/show/collections_list_component.html.erb

```diff
@@ -3,5 +3,5 @@
 <% collections.each do |collection| %>
   <%= render Elements::HeaderComponent.new(level: :h4, text: collection.title) %>
   <%= render Dashboard::Show::WorksListComponent.new(collection:) %>
-  <%= render Elements::ButtonLinkComponent.new(link: new_work_path(collection_id: collection.id), label: 'Deposit to this collection', variant: :primary) %>
+  <%= render Elements::ButtonLinkComponent.new(link: new_work_path(collection_druid: collection.druid), label: 'Deposit to this collection', variant: :primary) %>
```

Comment:
Switching to use collection druid as param instead of Collection id.

Reply:
This obviates the need to get the collection druid from the Collection when mapping.